### DISCUSSION TITLE
Struggling to style placeholder using tailwindcss in a project with vanilla javascript

### DISCUSSION THREAD

**Discussion by bkyerv** [ueberdosis/tiptap#3661] [General Questions] [answered]
I am building a project using **astro build** and **tailwind** and **tiptap**. I am struggling to style the placeholder because unlike in the example suggested in the docs where styles are applied using separate stylesheet I am using utility classes that must be passed to the elements `class` attributes. However I don't know how I can get access to `class` attribute of `placeholder` element. Any attempt to use `attribute` or `HTMLattribute` property of the Placeholder extension's configuaration object results in an error because such property doesn't exist, i.e. it is not possible set `attribute/HTMLattribute` property as in the case of the element containing the editor. I was wondering has someone faced this issue and if so how such issue is best resolved?

**Comment by tino-brst** [comment, marked as answer]
Kinda late, but for anyone else trying to do the same, you can make it work by making use of the `emptyEditorClass` prop on `Placeholder.configure`, and pass-in the Tailwind styles for the placeholder element. Using as example the [styles from the docs](https://tiptap.dev/api/extensions/placeholder#additional-setup):

```css
.tiptap p.is-editor-empty:first-child::before {
  color: #adb5bd;
  content: attr(data-placeholder);
  float: left;
  height: 0;
  pointer-events: none;
}
```

The same can be achieved via Tailwind like so:

```ts
Placeholder.configure({
  emptyEditorClass: "before:content-[attr(data-placeholder)] before:float-left before:text-[#adb5bd] before:h-0 before:pointer-events-none",
  placeholder: "My Custom Placeholder",
}),
```

Hope it helps!

**Reply by iqqmuT** [reply]
If you press enter in the empty editor, the placeholder will appear on the next line. To make Tailwind classes closer to the styles from the docs, add `first:` pseudo class:

```js
Placeholder.configure({
  emptyEditorClass: "first:before:content-[attr(data-placeholder)] first:before:float-left first:before:text-[#adb5bd] first:before:h-0 first:before:pointer-events-none",
  placeholder: "My Custom Placeholder",
}),
```
